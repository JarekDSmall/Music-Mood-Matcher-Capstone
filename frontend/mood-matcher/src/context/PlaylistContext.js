import React, { createContext, useState, useContext } from 'react';

const PlaylistContext = createContext();

export const PlaylistProvider = ({ children }) => {
    const [playlists, setPlaylists] = useState([]);

    const addPlaylist = (playlist) => {
        setPlaylists(prevPlaylists => [...prevPlaylists, playlist]);
    };

    const removePlaylist = (playlistId) => {
        setPlaylists(prevPlaylists => prevPlaylists.filter(pl => pl.id !== playlistId));
    };

    return (
        <PlaylistContext.Provider value={{ playlists, setPlaylists, addPlaylist, removePlaylist }}>
            {children}
        </PlaylistContext.Provider>
    );
};

export const usePlaylist = () => {
    const context = useContext(PlaylistContext);
    if (!context) {
        throw new Error('usePlaylist must be used within a PlaylistProvider');
    }
    return context;
};
