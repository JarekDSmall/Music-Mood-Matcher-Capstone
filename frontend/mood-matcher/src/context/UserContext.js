import React, { createContext, useState, useContext } from 'react';

const UserContext = createContext();

export const UserProvider = ({ children }) => {
    const [user, setUser] = useState(null);
    const [isAuthenticated, setIsAuthenticated] = useState(false);

    return (
        <UserContext.Provider value={{ user, setUser, isAuthenticated, setIsAuthenticated }}>
            {children}
        </UserContext.Provider>
    );
};

export const useUser = () => {
    const context = useContext(UserContext);
    if (!context) {
        throw new Error('useUser must be used within a UserProvider');
    }
    return context;
};
