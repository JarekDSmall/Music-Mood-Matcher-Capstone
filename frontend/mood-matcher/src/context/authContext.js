import React, { createContext, useContext, useState } from 'react';

export const AuthContext = createContext();

export const useAuth = () => {
    return useContext(AuthContext);
};

export const AuthProvider = ({ children }) => {
    const [currentUser, setCurrentUser] = useState(null);
    const [isAuthenticated, setIsAuthenticated] = useState(false);

    const login = (user) => {
        setCurrentUser(user);
        setIsAuthenticated(true);
    };

    const logout = () => {
        setCurrentUser(null);
        setIsAuthenticated(false);
    };

    const register = async (email, password) => {
        // Here, you'll typically call your backend API to register the user.
        // For now, I'll just simulate a successful registration.
        setCurrentUser({ email }); // This is just a mock. Replace with actual user data from your backend.
        setIsAuthenticated(true);
    };

    const contextValue = {
        currentUser,
        isAuthenticated,
        login,
        logout,
        register  // <-- Added the register function here
    };

    return (
        <AuthContext.Provider value={contextValue}>
            {children}
        </AuthContext.Provider>
    );
};

