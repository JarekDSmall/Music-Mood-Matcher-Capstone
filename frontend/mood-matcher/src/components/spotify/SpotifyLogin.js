import React, { useEffect, useContext } from 'react';
import axios from 'axios';
import { AuthContext } from '../../context/authContext';

function SpotifyLogin() {
  const { login } = useContext(AuthContext);

  useEffect(() => {
    const urlParams = new URLSearchParams(window.location.search);
    const code = urlParams.get('code');

    if (code) {
      const fetchAccessToken = async () => {
        try {
          const response = await axios.get(`/spotify/callback?code=${code}`);
          const { token } = response.data;
          if (token) {
            login(token);  // Assuming your login function accepts the token as a parameter
          }
        } catch (error) {
          console.error("Error fetching access token:", error);
        }
      };

      fetchAccessToken();
    }
  }, [login]);

  const handleLogin = async () => {
    try {
      const response = await axios.get('/spotify/auth-url');
      const authUrl = response.data.authUrl;
      window.location = authUrl;
    } catch (error) {
      console.error("Error fetching Spotify auth URL:", error);
    }
  };

  return (
    <div className="spotify-login-container">
      <h2>Login with Spotify</h2>
      <button className="spotify-login-button" onClick={handleLogin}>
        Login to Spotify
      </button>
    </div>
  );
}

export default SpotifyLogin;
