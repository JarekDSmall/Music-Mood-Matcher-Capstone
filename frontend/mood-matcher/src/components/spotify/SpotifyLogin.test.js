import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import SpotifyLogin from './SpotifyLogin';
import renderer from 'react-test-renderer';
import { AuthContext } from '../../context/authContext';

// Mock the context
const mockLogin = jest.fn();
const mockAuthContext = {
  login: mockLogin,
  isAuthenticated: false,
  // Add other properties and methods as needed by SpotifyLogin component
};

describe('SpotifyLogin Component', () => {
  it('matches snapshot', () => {
    const tree = renderer
      .create(
        <Router>
          <AuthContext.Provider value={mockAuthContext}>
            <SpotifyLogin />
          </AuthContext.Provider>
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  // ... other tests
});
