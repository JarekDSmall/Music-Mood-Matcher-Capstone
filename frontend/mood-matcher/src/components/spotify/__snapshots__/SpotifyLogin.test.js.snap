// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpotifyLogin Component matches snapshot 1`] = `
<div
  className="spotify-login-container"
>
  <h2>
    Login with Spotify
  </h2>
  <button
    className="spotify-login-button"
    onClick={[Function]}
  >
    Login to Spotify
  </button>
</div>
`;
